modules/mobile/app/models/mobile/v0/adapters/vaos_v2_appointment.rb

```diff
@@ -317,100 +321,111 @@ def appointment_type
-                  zip_code: cc_location.dig(:address, :postal_code)
-                }
-                if cc_location[:telecom].present?
-                  phone_number = cc_location[:telecom]&.find do |contact|
-                    contact[:system] == CONTACT_TYPE[:phone]
-                  end&.dig(:value)
-
-                  location[:phone] = parse_phone(phone_number)
-                end
-
-              end
+              appointment_request? ? set_cc_appointment_request_location : set_cc_appointment_location
             when APPOINTMENT_TYPES[:va_video_connect_atlas],
               APPOINTMENT_TYPES[:va_video_connect_home],
               APPOINTMENT_TYPES[:va_video_connect_gfe]
-
-              location[:name] = appointment.dig(:location, :name)
-              location[:phone] = parse_phone(appointment.dig(:location, :phone, :main))
-              telehealth = appointment[:telehealth]
-
-              if telehealth
-                address = telehealth.dig(:atlas, :address)
-
-                if address
-                  location[:address] = {
-                    street: address[:street_address],
-                    city: address[:city],
-                    state: address[:state],
-                    zip_code: address[:zip_code],
-                    country: address[:country]
-                  }
-                end
-                location[:url] = telehealth[:url]
-                location[:code] = telehealth.dig(:atlas, :confirmation_code)
-              end
+              set_telehealth_location
             else
-              location[:id] = appointment.dig(:location, :id)
-              location[:name] = appointment.dig(:location, :name)
-
-              address = appointment.dig(:location, :physical_address)
-              if address.present?
-                location[:address] = {
-                  street: address[:line]&.join(' ')&.strip,
-                  city: address[:city],
-                  state: address[:state],
-                  zip_code: address[:postal_code]
-                }
-              end
-              location[:lat] = appointment.dig(:location, :lat)
-              location[:long] = appointment.dig(:location, :long)
-              location[:phone] = parse_phone(appointment.dig(:location, :phone, :main))
+              set_va_appointment_location
             end
 
-            location
+            location_template
           end
         end
-        # rubocop:enable Metrics/MethodLength
+
+        def location_template
+          @location_template ||= {
+            id: nil,
+            name: nil,
+            address: {
+              street: nil,
+              city: nil,
+              state: nil,
+              zip_code: nil
+            },
+            lat: nil,
+            long: nil,
+            phone: {
+              area_code: nil,
+              number: nil,
+              extension: nil
+            },
+            url: nil,
+            code: nil
+          }
+        end
+
```

Comment:
This method is the only new functionality here. The rest is just breaking the too long method down into smaller pieces and restructuring the code to avoid too many nested structures. I got this functionality from going through the front end web code and confirming behavior with Simi.